util/dvsim/NcLauncher.py

```diff
@@ -86,49 +85,49 @@ def _do_launch(self):
 
         self._dump_env_vars(exports)
 
+        # using os.open instead of fopen as this allows
+        # sharing of file descriptors across processes
+        fd = os.open(self.deploy.get_log_path(), os.O_WRONLY | os.O_CREAT)
+        os.set_inheritable(fd, True)
+        message = '[Executing]:\n{}\n\n'.format(self.deploy.cmd)
+        byte_message = message.encode('utf-8')
+        os.write(fd, byte_message)
+
         if self.deploy.sim_cfg.interactive:
             # Interactive: Set RUN_INTERACTIVE to 1
             exports['RUN_INTERACTIVE'] = '1'
-            interactive_flags = ['-I']
-            cmd_arr = self.get_submit_cmd(interactive_flags)
-            log.log(VERBOSE, '[Executing]:\n{}\n\n'.format(self.deploy.cmd))
+            cmd_arr = self.get_submit_cmd(['-I'])
+            buf_size = 1
```

Comment:
I think this is a change and we're now doing line buffering? That might be reasonable, but maybe it makes sense to leave a note explaining why?

Reply:
done. added a comment